### ISSUE TITLE
appdash go.mod import is invalid

### ISSUE THREAD

**Issue by AdallomRoy** [pulumi/pulumi#14725]
### What happened?

Pulumi doesn't build because of an invalid reference
`        github.com/pulumi/pulumi/sdk/v3/go/pulumi imports
        github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil imports
        github.com/pulumi/appdash/opentracing: github.com/pulumi/appdash@v0.0.0-20231130102013-538fec2a741d: invalid version: unknown revision 538fec2a741d
`

### Example

Pulumi doesn't build because of an invalid reference
`        github.com/pulumi/pulumi/sdk/v3/go/pulumi imports
        github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil imports
        github.com/pulumi/appdash/opentracing: github.com/pulumi/appdash@v0.0.0-20231130102013-538fec2a741d: invalid version: unknown revision 538fec2a741d

### Output of `pulumi about`

github.com/pulumi/pulumi/sdk/v3 v3.95.0

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by AdallomRoy** [comment]
Unfortunately, there are still many projects within the pulumi ecosystem that depend on the old non-existent appdash repo:
`github.com/pulumi/esc@v0.5.6 sourcegraph.com/sourcegraph/appdash@v0.0.0-20211028080628-e2786a622600
github.com/pulumi/pulumi-random/sdk/v4@v4.14.0 sourcegraph.com/sourcegraph/appdash@v0.0.0-20211028080628-e2786a622600
github.com/pulumi/pulumi-tls/sdk/v4@v4.11.1 sourcegraph.com/sourcegraph/appdash@v0.0.0-20211028080628-e2786a622600
`

I think you should release tags to all of these in order to completely remove the usage.

**Comment by Frassle** [comment]
cc @pulumi/providers for the above

**Comment by AdallomRoy** [comment]
I don't think they were actually tagged. Do you prefer that I open a new ticket somewhere?

**Comment by lukehoban** [comment]
@AdallomRoy Thanks for raising and highlighting the additional downstream places that need updates.  We are working through this right now.  As you might imagine, this is also impacting our own build and release processes.

For some additional context - there are two Go modules we publish out of `pulumi/pulumi` - `sdk` and `pkg`.  

The changes in #14726 fixed this for the `sdk` module, which was the one you highlighted in the original note above in this issue.

However, we have not *yet* been able to address this for the `pkg` module.  Unfortunately, we have a transitive dependency through our gocloud.dev dependency on this package, so need to update both of those.  But the gocloud.dev dependency has caused us some non-trivial breaking changes when we've tried to adopt it, so we've been stuck on an older version.  We are working through a plan to safely move to the latest versions of that dependency to unblock getting our `pkg` module onto a "correct" version of appdash.

For downstream dependencies which only rely on `sdk`, this issue is fixed.

For downstream dependencies which depend on both, we are not yet able to update them.  

* `github.com/pulumi/esc` has dependencies on both, so upgrade is blocked still.
* `github.com/pulumi/pulumi-random/sdk/v4` only depends on `sdk` and was fixed in the latest release https://github.com/pulumi/pulumi-random/releases/tag/v4.15.0
*  `github.com/pulumi/pulumi-tls/sdk/v4` was fixed in https://github.com/pulumi/pulumi-tls/pull/364 and a new release should be tagged shortly.

If you can share any details, would love to understand how you are building these packages yourselves, so that we can offer potential workaround ahead of a full unblocking of the dependency issues here (or work to ensure we are testing against your build structure in our pipelines).
